prompting/validators/tasks.py

```diff
@@ -0,0 +1,138 @@
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent("""\
+        Your task is to summarize the text delimited with triple backticks:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}
+        - Do not try to create questions or answers for your summarization. 
+        """)
+
+        prompt = prompt_template.format(base_text=self.base_text, criteria=criteria_bullet_points_str)        
+        return prompt
+
+class QuestionGenerationTask(Task):
+    def validate(self, responses: List[str]) -> torch.FloatTensor:
+        # The logic for validation is the same as the base class, so we can call the base class's method.
+        return super().validate(responses)
+
+    def compose_prompt(self) -> str:
+        # Aggregates criteria in bullet points
+        criteria_bullet_points = [f"- {criterion.compose_text()}" for criterion in self.criteria]
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent("""\
+        Your task is to ask a single relevant and insightful question about the preceding context delimited with triple backticks:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}                                      
+        - Do not answer the question you generate. 
+        - Do not try to summarize the text
+        """)
+
+        prompt = prompt_template.format(base_text=self.base_text, criteria=criteria_bullet_points_str)        
+        return prompt
+
+class QuestionAnswerTask(Task):
+    def validate(self, responses: List[str]) -> torch.FloatTensor:
+        # The logic for validation is the same as the base class, so we can call the base class's method.
+        return super().validate(responses)
+        
+    def compose_prompt(self) -> str:
+        # Aggregates criteria in bullet points
+        criteria_bullet_points = [f"- {criterion.compose_text()}" for criterion in self.criteria]
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent("""\
+        Read the preceding context delimited with triple backticks carefully. Your task is to answer the question step by step and explain your thoughts:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}                                      
+        - Do not include questions or summaries in your answer.
+        """)
+
+        prompt = prompt_template.format(base_text=self.base_text, criteria=criteria_bullet_points_str)        
+        return prompt
+
+
+def create_summarization_task(base_text: str) -> SummaryTask:
+    criteria = [
+        MaxOfNWordsCriteria(n_words=random.randint(50, 200), penalty=0.1),
+        MatchNSentencesCriteria(n_sentences=random.randint(4, 8), penalty=0.1)
```

Comment:
We could replace `MatchNSentencesCriteria` with `MatchLengthCriteria` which supports different units such as `['chars', 'words', 'sentences', 'paragraphs']`